### PR TITLE
[Merged by Bors] - chore: stop using 'old-style' `tfae_have` syntax

### PR DESCRIPTION
**Pull request by thorimur**
Banishes "Mathlib `have`" style syntax for `tfae_have` (but does not officially deprecate it). Since we now have the chance to supply a term after `tfae_have ... :=`, we remove one-line `exact`s where possible, and convert `intro`s and some `rintros` into match alternatives. Most single-line `fun`s are preserved.

---

This PR should be merged just before #17045, which deprecates the syntax via warning.

- [x] depends on: #10653 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/RCLike/Basic.lean [hunk 1]
```diff
@@ -317,16 +317,14 @@ open List in
...
   tfae_have 4 → 3
-  · intro h
+  | h => by
     conv_rhs => rw [← re_add_im z, h, ofReal_zero, zero_mul, add_zero]
-  tfae_have 3 → 2
-  · exact fun h => ⟨_, h⟩
-  tfae_have 2 → 1
-  · exact fun ⟨r, hr⟩ => hr ▸ conj_ofReal _
+  tfae_have 3 → 2 := fun h => ⟨_, h⟩
+  tfae_have 2 → 1 := fun ⟨r, hr⟩ => hr ▸ conj_ofReal _
```
Comment (joneugster) [inline Mathlib/Analysis/RCLike/Basic.lean:327]:
```suggestion
  tfae_have 3 → 2
  | h => ⟨_, h⟩
  tfae_have 2 → 1
  | ⟨r, hr⟩ => hr ▸ conj_ofReal _
```

You could, if you wanted, search-replace (regex) `tfae_have.*:=[\s\n]*fun` with the `| ... =>` pattern. It's also fine if you decide to leave as-is for now.

**Review by joneugster — APPROVED** [review]

File: Mathlib/Analysis/RCLike/Basic.lean (hunk 1 above)
Comment (thorimur) [reply Mathlib/Analysis/RCLike/Basic.lean:327]: Generally, my policy has been to maintain single-line `fun`s (with term-mode bodies), or match the surrounding style of `tfae_have` if there was one (e.g., all match alts, or all `:=`). 

However, note that another option is keeping things like this single-line with e.g. 
```
tfae_have 3 → 2 | h => ⟨_, h⟩ 
tfae_have 2 → 1 | ⟨r, hr⟩ => hr ▸ conj_ofReal _  
```
It is easy via regex, so if you (or zulip) have a strong opinion on this style vs. `:= fun` style, I'm happy to change it! :)

Comment (joneugster) [reply Mathlib/Analysis/RCLike/Basic.lean:327]: I don't have a strong opinion

**Review by joneugster — APPROVED** [review]
Thanks for the PR!

I've looked through all changes, and indeed all of them seem to be purely stylistic changes removing the old `have_tfae` syntax. This opens the door for the deprecation warning following in #17045

maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**